eq-author-api/schema/resolvers/logic/binaryExpression2/index.js

```diff
@@ -168,9 +168,15 @@ Resolvers.Mutation = {
     };
     delete updatedLeftSide.nullReason;
 
+    const getRightSide = {
+      ...expression.right,
+    };
+
     expression.left = updatedLeftSide;
-    expression.right = null;
-    expression.condition = answerTypeToConditions.getDefault(answer.type);
+    expression.right = getRightSide;
+    expression.condition = expression.condition
+      ? expression.condition
+      : answerTypeToConditions.getDefault(answer.type);
```

Comment:
```suggestion
    expression.condition = expression.condition ?? answerTypeToConditions.getDefault(answer.type);
```

This change is called "Nullish coalescing operator":
- If the left side condition (`expression.condition`) is not null, it returns the left side (`expression.condition`).
- If the left side condition (`expression.condition`) is null, it returns the right side (`answerTypeToConditions.getDefault(answer.type)`)

This change will return the same result as the original code

Reply:
Nice Sam!